js/kadai.js

```diff
@@ -0,0 +1,132 @@
+      results[4].incorrect_answers);
+    const quiz_six = new Quiz(results[5].category,results[5].difficulty,results[5].question,results[5].correct_answer,
+      results[5].incorrect_answers);
+    const quiz_seven = new Quiz(results[6].category,results[6].difficulty,results[6].question,results[6].correct_answer,
+      results[6].incorrect_answers);
+    const quiz_eight = new Quiz(results[7].category,results[7].difficulty,results[7].question,results[7].correct_answer,
+      results[7].incorrect_answers);
+    const quiz_nine = new Quiz(results[8].category,results[8].difficulty,results[8].question,results[8].correct_answer,
+      results[8].incorrect_answers);
+    const quiz_ten = new Quiz(results[9].category,results[9].difficulty,results[9].question,results[9].correct_answer,
+      results[9].incorrect_answers);
+    const quizData = [quiz_one,quiz_two,quiz_three,quiz_four,quiz_five,quiz_six,quiz_seven,quiz_eight,quiz_nine,quiz_ten]
+    startQuiz(quizData)
+  })
+
+  const startQuiz = async (quizData) => {
+    genre.textContent = '[ジャンル]' + quizData[quizCount].genre
+    genre.style.fontWeight = "bold";
+    genre.style.fontsize = "1rem"
+    difficulty.textContent = '[難易度]' + quizData[quizCount].difficulty
+    difficulty.style.fontWeight = "bold";
+    titleText.after(genre)
+    genre.after(difficulty)
+    titleText.textContent = '問題'
+    descriptionText.textContent = quizData[quizCount].question
+    for(let i = 0; i < quizData[quizCount].choiceList.length; i++){
+      if(typeof quizData[quizCount].choiceList[i] == 'undefined'){
+        break
+      }
+      let choiceButton = document.createElement('button')
+      calcAnswer(choiceButton,quizData)
+      let p = document.createElement('p')
+      choiceButton.textContent = quizData[quizCount].choiceList[i]
+      bottomLine.after(answerBox)
+      answerBox.appendChild(p)
+      p.appendChild(choiceButton)
+    }
+  }
+  const calcAnswer = (choiceButton,quizData) => {
+    choiceButton.addEventListener('click', () => {
+      // console.log('選択ボタンクリック')
+      if (choiceButton.textContent == quizData[quizCount].correctAnswer){
+        numberOfCorrectAnswers++
+      }
+      quizCount += 1
+      while( answerBox.firstChild ){
+        answerBox.removeChild(answerBox.firstChild );
+      }
+      genre.remove()
+      difficulty.remove()
+      if (resultLength == quizCount){
+        titleText.textContent = "あなたの正当数は" + numberOfCorrectAnswers + "です"
+        descriptionText.textContent = "再度チャレンジしたい場合は以下のボタンをクリック！"
+        const homeButton = document.createElement("button")
+        homeButton.textContent = "ホームに戻る"
+        answerBox.appendChild(homeButton)
+        homeButton.addEventListener("click", () => {
+          location.reload();
+        })
+      }
+      startQuiz(quizData)
+    })
+  }
+  const getQuizData = async () => {
+    const respons = await fetch('https://opentdb.com/api.php?amount=10')
+    const data = await respons.json();
```

Comment:
Api 呼び出しをするところは `try/catch` を利用することを心掛けてください。
Api はネットワークエラーなど予期しないエラーが発生しやすいです。

```suggestion
    try {
      const respons = await fetch('https://opentdb.com/api.php?amount=10')
      const data = await respons.json();
      return data;
    } catch (e) {
      console.error(e)
    }
```